_health-care/_js/_components/_health-care-app.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import { Nav } from './_hello-world.jsx';
+
+class HealthCareApp extends React.Component {
+  constructor() {
+    super();
+    this.onStateChange = this.onStateChange.bind(this);
+    this.state = { applicationData: {} };
+    const savedAppData = window.localStorage["health-app-data"];
+    if (savedAppData) {
+      this.state.applicationData = JSON.parse(savedAppData);
+    }
+    if (this.state.applicationData.sharedText === undefined) {
+      this.state.applicationData.sharedText = "shared";
+    }
+    if (this.state.applicationData.happyDate === undefined) {
+      this.state.applicationData.happyDate = {
+        month: "1",
+        day: "15",
+        year: "1997",
+      };
+    }
+    if (this.state.applicationData.sadDate === undefined) {
+      this.state.applicationData.sadDate = {
+        month: "4",
+        day: "5",
+        year: "2014",
+      };
+    }
+  }
+
+  onStateChange(field, update) {
+    let newApplicationData = Object.assign({}, this.state.applicationData);
+    newApplicationData[field] = update;
+    window.localStorage["health-app-data"] = JSON.stringify(newApplicationData);
+
+    this.setState( {applicationData: newApplicationData} );
+  }
+
+  render()  {
+    if (this === null) {console.log("huh"); }
+    return (
+      <div>
+        <h1>Top-Level of App</h1>
+        <p> Shared text is: { this.state.applicationData.sharedText } </p>
+        <Nav/>
+        {
+          // Pass the application state down to child components. The
+          // cloneElement idiom is the standard react way to add
+          // properties to a child. It doesn't actually new a bunch of objects
+          // so it's fast.
+          React.Children.map(this.props.children, (child) => {
+              return React.cloneElement(child,
+                { applicationData: this.state.applicationData,
+                  onStateChange: this.onStateChange })
```

Comment:
Correctish.

I assume each Page will more explicitly pass a subset of structure down to their Question component which will then filter down into the input fields.

It doesn't all quite work yet and is clunky, but this is indeed how data is supposed to propagate.

Another option is to use `window.applicationData`, but that's a global variable and I hate globals.  (Yes...applicationData is also a global variable effectively, but the `setState()` + `props` patterns restrict mutation and inspection so it's actually easier to reason about still.)